Gems/WhiteBox/Code/Source/Viewport/WhiteBoxVertexTranslationModifier.cpp

```diff
@@ -208,6 +208,17 @@ namespace WhiteBox
                 Api::CalculatePlanarUVs(*whiteBox);
             });
 
+        m_translationManipulator->InstallInvalidateCallback(
+            [this, sharedState]()
+            {
+                WhiteBoxMesh* whiteBox = nullptr;
+                EditorWhiteBoxComponentRequestBus::EventResult(
+                    whiteBox, m_entityComponentIdPair, &EditorWhiteBoxComponentRequests::GetWhiteBoxMesh);
+                EditorWhiteBoxComponentRequestBus::Event(m_entityComponentIdPair, &EditorWhiteBoxComponentRequests::DeserializeWhiteBox);
+                
+                this->AZ::TickBus::Handler::BusDisconnect();
+            });
```

Comment:
question: Just curious do we definitely need this? What happens if this isn't here?

Reply:
calling DeserializeWhiteBox just restores the mesh before the operation. just undoes the operation when you switching from default to another mode in the middle of an operation.

Reply:
Aaah I see, makes sense thanks :+1: